keras_nlp/models/roberta/roberta_masked_lm_preprocessor_test.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for RoBERTa masked language model preprocessor layer."""
+
+import os
+
+import tensorflow as tf
+from absl.testing import parameterized
+from tensorflow import keras
+
+from keras_nlp.models.roberta.roberta_masked_lm_preprocessor import (
+    RobertaMaskedLMPreprocessor,
+)
+from keras_nlp.models.roberta.roberta_tokenizer import RobertaTokenizer
+
+
+class RobertaMaskedLMPreprocessorTest(tf.test.TestCase, parameterized.TestCase):
+    def setUp(self):
+        vocab = {
+            "<s>": 0,
+            "<pad>": 1,
+            "</s>": 2,
+            "Ġair": 3,
+            "plane": 4,
+            "Ġat": 5,
+            "port": 6,
+            "Ġkoh": 7,
+            "li": 8,
+            "Ġis": 9,
+            "Ġthe": 10,
+            "Ġbest": 11,
+            "<mask>": 12,
+        }
+
+        merges = ["Ġ a", "Ġ t", "Ġ k", "Ġ i", "Ġ b", "Ġa i", "p l", "n e"]
+        merges += ["Ġa t", "p o", "r t", "o h", "l i", "Ġi s", "Ġb e", "s t"]
+        merges += ["Ġt h", "Ġai r", "pl a", "Ġk oh", "Ġth e", "Ġbe st", "po rt"]
+        merges += ["pla ne"]
+
+        self.preprocessor = RobertaMaskedLMPreprocessor(
+            tokenizer=RobertaTokenizer(
+                vocabulary=vocab,
+                merges=merges,
+            ),
+            mask_selection_rate=0.3,
+            mask_selection_length=4,
+            sequence_length=12,
+        )
+        keras.utils.set_random_seed(42)
```

Comment:
Is it good practice to assert random output as a function of a random seed? My understanding is that standard process is to mock the function with random output and then just have an overall integration test.

Reply:
Let me poke around. The thing that generates the random output is kinda complex down in `MaskedLMMaskGenerator`, and when mapping with tf.data the whole call graph will be compiled. Mocking might be more trouble than it's worth, and only apply in the note compiled case.

In [our tests there](https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/layers/masked_lm_mask_generator_test.py) we just do a lot of shape assertions, and don't assert the exact structure. Maybe that is the move.

Regardless, I am down to remove the random seed setting.

Reply:
OK, got some more deterministic seedless testing in. Did not go with the mock as I think that would have ended up being pretty fragile.

I also exposed a few more of our underlying "mask generator" options of the preprocessing layer. I think they will be useful to have (they certainly are for these tests).